lib/runtime.cpp

```diff
@@ -0,0 +1,228 @@
+                                               std::placeholders::_1, std::placeholders::_2);
+        this->callbacks.register_ext_mqtt_subscribe_handler(external_mqtt_handler);
+
+        // FIXME (aw): would be nice to move this config related thing toward the module_init function
+        std::vector<cmd> cmds =
+            this->callbacks.everest_register(config.get_main_config()[this->module_id]["connections"]);
+
+        for (auto const& command : cmds) {
+            everest.provide_cmd(command);
+        }
+
+        auto module_configs = config.get_module_configs(this->module_id);
+        const auto module_info = config.get_module_info(this->module_id);
+
+        this->callbacks.init(module_configs, module_info);
+
+        std::thread mainloop_thread = std::thread(&Everest::Everest::mainloop, &everest);
+
+        // register the modules ready handler with the framework
+        // this handler gets called when the global ready signal is received
+        everest.register_on_ready_handler(this->callbacks.ready);
+
+        // the module should now be ready
+        everest.signal_ready();
+
+        mainloop_thread.join();
+
+        EVLOG(info) << "Exiting...";
+    } catch (boost::exception& e) {
+        EVLOG(critical) << fmt::format("Caught top level boost::exception:\n{}",
+                                       boost::diagnostic_information(e, true));
+    } catch (std::exception& e) {
+        EVLOG(critical) << fmt::format("Caught top level std::exception:\n{}", boost::diagnostic_information(e, true));
+    }
+
+    return 0;
+}
+
+bool ModuleLoader::parse_command_line(int argc, char* argv[]) {
+    po::options_description desc("EVerest");
+    desc.add_options()("help,h", "produce help message");
+    desc.add_options()("main_dir", po::value<std::string>(), "Set main EVerest directory");
+    desc.add_options()("module,m", po::value<std::string>(),
+                       "Which module should be executed (module id from config file)");
+    desc.add_options()("log_conf", po::value<std::string>(), "The path to a custom logging.ini");
+    desc.add_options()("dontvalidateschema", "Don't validate json schema on every message");
+    desc.add_options()("conf", po::value<std::string>(), "The path to a custom config.json");
+
+    po::variables_map vm;
+    po::store(po::parse_command_line(argc, argv, desc), vm);
+    po::notify(vm);
+
+    if (vm.count("help") != 0) {
+        std::cout << desc << "\n";
+        return false;
+    }
+
+    if (vm.count("main_dir") == 0) {
+        EVTHROW(EVEXCEPTION(EverestApiError, "--main_dir parameter is required"));
+    }
+
+    this->runtime_settings = std::make_unique<RuntimeSettings>(vm);
+    this->original_process_name = argv[0];
+
+    if (vm.count("module") != 0) {
+        this->module_id = vm["module"].as<std::string>();
+    } else {
+        EVTHROW(EVEXCEPTION(EverestApiError, "--module parameter is required"));
+    }
+
+    return true;
```

Comment:
Although this gives additional flexibility, I don't really like the fact, that the manager has already parsed all the arguments, just to forward them and let them parsed again.  Something related to this would also be, that the spawned modules shouldn't know about the complete configuration - shouldn't they only get their module and implementation specific config (which could be passed by the manager via mqtt?)

Reply:
I like the idea of removing the double-parsing of the config a lot, but since this will probably be a bigger change I'll tackle that in a dedicated PR